client/src/featureform/resources.py

```diff
@@ -771,19 +807,25 @@ def _check_insert_training_set_resources(self, db) -> None:
             db.getNameVariant("label_variant", "name", self.label[0], "variant", self.label[1])
         except ValueError:
             raise ValueError("{} does not exist. Failed to register training set".format(self.label[0]))
-        for feature in self.features:
+        for feature_name, feature_variant in self.features:
             try:
-                db.getNameVariant("feature_variant", "name", feature[0], "variant", feature[1])
+                db.getNameVariant("feature_variant", "name", feature_name, "variant", feature_variant)
             except ValueError:
-                raise ValueError("{} does not exist. Failed to register training set".format(feature[0]))
+                raise ValueError("{} does not exist. Failed to register training set".format(feature_name))
```

Comment:
Would recommend using f-strings. 
```
f"{feature_name} does not exist. Failed to register training set"
``` 
It makes it easier to read when looking at the code.